plugins/woocommerce/src/Internal/Utilities/FilesystemUtil.php

```diff
@@ -121,4 +121,58 @@ protected static function initialize_wp_filesystem(): bool {
 
 		return is_null( $initialized ) ? false : $initialized;
 	}
+
+	/**
+	 * Validate that a file path is a valid upload path.
+	 *
+	 * @param string $path The path to validate.
+	 * @throws \Exception If the file path is not a valid upload path.
+	 */
+	public static function validate_upload_file_path( string $path ): void {
+		$pre = apply_filters( 'woocommerce_pre_validate_upload_file_path', false, $path );
```

Comment:
This filter was suggested in the original issue, but I was able to upload fine without it.  The primary issues were with the file path modification we do below.

Reply:
I was trying to match the existing `pre_x` filters that exist around WP, but we'll be removing this filter as this is confirmed to work fine without 👍